### ISSUE TITLE
`react/no-unused-prop-types` false positives with prop alias [TS]

### ISSUE THREAD

**Issue by awwit** [yannickcr/eslint-plugin-react#2686]
I have a similar code:

```ts
import * as React from 'react'

interface Props {
  'aria-label': string // 'undefined' PropType is defined but prop is never used eslint(react/no-unused-prop-types)
  // 'undefined' PropType is defined but prop is never used eslint(react-redux/no-unused-prop-types)
}

export default function Component({
  'aria-label': ariaLabel, // 'aria-label' is missing in props validation eslint(react/prop-types)
}: Props): JSX.Element {
  return <div aria-label={ariaLabel} />
}
```

I get a false positive on the `aria-label` property (and on any other property enclosed in quotation marks where non-standard characters are used).

eslint@7.3.1
eslint-plugin-react@7.20.2
eslint-plugin-react-redux@3.0.3

**Comment by hank121314** [comment]
This issue is caused by string literal.
eslint-plugin-react does not support string literal for interface key currently.
I will make a pr to fix it soon.

**Comment by ljharb** [comment]
v7.20.3 is released.
